src/Unity.Mathematics.Shaders/hlsl.cs

```diff
@@ -2,105 +2,367 @@
 
 namespace Unity.Mathematics.Shaders
 {
+    /// <summary>
+    ///
+    /// </summary>
     public static class hlsl
     {
+        /// <summary>
+        ///
+        /// </summary>
+        /// <param name="x"></param>
+        /// <returns></returns>
         [MethodImpl((MethodImplOptions)0x100)] // agressive inline
         public static float ddx(float x) { throw new System.NotImplementedException(); }
```

Comment:
I would just delete all these methods that do `throw new System.NotImplementedException();`. There is no user who would object to it.

Reply:
Forum post: OMG what happened ddy_coarse? I was using it to throw NonImplementedException. :D